service/storage_proxy: Add metrics keeping track of incoming hints

dawmd opened the pull request:
Although Scylla already exposes metrics keeping track of various information related to hinted handoff, all of them correspond to either storing or sending hints. However, when debugging, it's also crucial to be aware of how many hints are coming to a given node and what their size is. Unfortunately, the existing metrics are not enough to obtain that information.

This PR introduces the following new metrics:

* `sent_bytes_total` – the total size of the hints that have been sent from a given shard,
* `received_hints_total` – the total number of hints that a given shard has received,
* `received_hints_bytes_total` – the total size of the hints a given shard has received.

It also renames `hints_manager_sent` to `hints_manager_sent_total` to avoid conflicts of prefixes between that metric and `sent_bytes_total` in tests.

Fixes scylladb/scylladb#10987

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9334/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9334/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/9832/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9334/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/1810/)
#### Failed Tests (8/30513):
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9334/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9334/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9334/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9334/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9334/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_4) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9334/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_5) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9334/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_6) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9334/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest_with_topology_changes___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
#### Build Details:
- Duration: 4 hr 30 min
- Builder: spider1.cloudius-systems.com

dawmd commented:
The failing `test_fence_hints` is irrelevant – the test is skipped. The failure in `test_hintedhandoff_switch_config_in_runtime_via_http_api` is most likely caused by scylladb/scylladb#19006.

piodul commented on service/storage_proxy.cc:
```diff
@@ -2785,6 +2788,14 @@ void storage_proxy_stats::stats::register_stats() {
+
+        sm::make_total_bytes("received_hint_total_size", received_hint_total_size,
+                        sm::description("total size of hints received by this node"),
+                       {storage_proxy_stats::current_scheduling_group_label()}).set_skip_when_empty(false),
```
Do we need `.set_skip_when_empty(false)` here? I suppose that the metric will not be larger than 0 on other scheduling groups than `streaming`, so this is a worthwhile optimization. On the other hand, if there are some hints received in other scheduling groups then it will be a clear indication of a bug.

piodul commented on service/storage_proxy_stats.hh:
```diff
@@ -191,6 +191,10 @@ struct stats : public write_stats {
     split_stats mutation_data_read_errors;
 
+    // Received hints
+    size_t received_hint_count = 0;
+    size_t received_hint_total_size = 0; /// The size is in bytes.
```
Why `size_t` and not `uint64_t` like other metrics in this struct?

piodul commented on service/storage_proxy.cc:
```diff
@@ -2791,11 +2791,11 @@ void storage_proxy_stats::stats::register_stats() {
         sm::make_total_bytes("received_hint_total_size", received_hint_total_size,
                         sm::description("total size of hints received by this node"),
-                       {storage_proxy_stats::current_scheduling_group_label()}).set_skip_when_empty(false),
+                        {storage_proxy_stats::current_scheduling_group_label()}).set_skip_when_empty(),
```
Ah, you removed `.set_skip_when_empty(false)`, but you did it in the wrong commit.

piodul reviewed (commented):
Left some nitpicks, otherwise LGTM.

amnonh commented on db/hints/manager.cc:
```diff
@@ -174,6 +174,9 @@ void manager::register_metrics(const sstring& group_name) {
         sm::make_counter("sent", _stats.sent,
                         sm::description("Number of sent hints.")),
 
+        sm::make_gauge("sent_total_size", _stats.sent_hints_size,
```
Assuming that it's the total bytes ever sent, it should be counter (ever growing) and a better name would be sent_bytes_total

amnonh commented on service/storage_proxy.cc:
```diff
@@ -2785,6 +2788,14 @@ void storage_proxy_stats::stats::register_stats() {
                        sm::description("how many times a coordinator did not perform prune after cas"),
                        {storage_proxy_stats::current_scheduling_group_label()}).set_skip_when_empty(),
+
+        sm::make_total_operations("received_hint_count", received_hint_count,
```
we have deprecated collectd, use counter or gauge.
and a better name will be received_hints_total

amnonh commented on service/storage_proxy.cc:
```diff
@@ -2785,6 +2788,14 @@ void storage_proxy_stats::stats::register_stats() {
+                        sm::description("number of hints received by this node"),
+                        {storage_proxy_stats::current_scheduling_group_label()}).set_skip_when_empty(),
+
+        sm::make_total_bytes("received_hint_total_size", received_hint_total_size,
```
use counter,  a better name will be received_hints_bytes_total

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9484/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9484/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/1900/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/9924/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9484/)
#### Failed Tests (9/30654):
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9484/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest_with_topology_changes___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9484/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [test_change_two](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9484/testReport/junit/%28root%29/test_change_ip/Tests___Unit_Tests___test_change_two) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_change_two)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9484/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9484/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9484/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9484/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_4) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9484/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_5) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9484/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_6) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
#### Build Details:
- Duration: 5 hr 1 min
- Builder: i-07237347fd9813a62 (m5ad.12xlarge)

dawmd commented:
Version 2–4:
* rebased on top of the current master
* renamed `sent_hints_size` to `sent_hints_bytes_total`, `sent_total_size` to `sent_bytes_total`, `received_hint_total_size` to `received_hints_bytes_total`
* included information that incoming hints are both regular hints and MV hints in the description of the metrics in storage proxy
* started using counter instead of gauge/total_bytes

amnonh commented on db/hints/manager.cc:
```diff
@@ -174,6 +174,9 @@ void manager::register_metrics(const sstring& group_name) {
         sm::make_counter("sent", _stats.sent,
                         sm::description("Number of sent hints.")),
 
+        sm::make_counter("sent_bytes_total", _stats.sent_hints_bytes_total,
```
Why do the receive hints have skip_when_empty, and they sent don't?

dawmd replied:
That's a good point. None of the hinted handoff metrics use `skip_when_empty`. I'm not knowledgeable in how exactly it's used (please, correct me if I'm wrong), but the way I understand it is the metrics won't be "displayed" unless they've been modified at least once.

In the context of hints, I think there's a merit in keeping them that way – when analysing the behaviour of a node, hints often play a significant role in how performant it is, so seeing no hints have been sent or written is useful because it gives you that information right away. The same reasoning could be applied to the metrics with incoming hints (which would mean I shouldn't have made them skipped when empty). But I'm sure it's still possible to obtain those metrics, so perhaps it'd be better to skip them when empty anyway. I definitely agree the metrics corresponding to errors, etc. should be skipped.

amnonh commented on service/storage_proxy.cc:
```diff
@@ -2807,6 +2810,14 @@ void storage_proxy_stats::stats::register_stats() {
                        sm::description("how many times a coordinator did not perform prune after cas"),
                        {storage_proxy_stats::current_scheduling_group_label()}).set_skip_when_empty(),
+
+        sm::make_counter("received_hint_count", received_hint_count,
```
received_hints_total

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9579/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9579/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/1984/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10009/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9579/)
#### Failed Tests (8/30719):
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9579/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest_with_topology_changes___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9579/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9579/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9579/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9579/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9579/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_4) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9579/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_5) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9579/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_6) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
#### Build Details:
- Duration: 2 hr 46 min
- Builder: spider2.cloudius-systems.com

dawmd commented:
Version 5:
* `received_hint_count` renamed to `received_hints_total`

amnonh reviewed (approved):
LGTM (just update the message to reflect the updated names)

dawmd commented:
The PR message updated

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9630/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9630/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10051/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2028/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9630/)
#### Failed Tests (8/30727):
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9630/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9630/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest_with_topology_changes___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9630/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9630/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9630/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9630/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_4) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9630/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_5) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
- [test_fence_hints](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9630/testReport/junit/%28root%29/test_fencing/Tests___Unit_Tests___test_fence_hints_6) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_fence_hints)
#### Build Details:
- Duration: 2 hr 38 min
- Builder: spider2.cloudius-systems.com

dawmd commented:
Version 6:
* fixed the failing test: `topology_experimental_raft/test_fencing` (`test_fence_hints`)
* updated the commit message accordingly

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9643/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9643/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: topology_experimental_raft/test_fencing
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9643/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2033/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10056/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9643/)
#### Failed Tests (2/31325):
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9643/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest_with_topology_changes___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9643/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
#### Build Details:
- Duration: 3 hr 1 min
- Builder: spider8.cloudius-systems.com

dawmd commented:
The failing test is scylladb/scylladb#19006. These changes should have no impact on it

mykaul commented:
@dawmd - some maintainers ask to rerun CI. Please do.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9710/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9710/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: topology_experimental_raft/test_fencing
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9710/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2088/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10111/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9710/)
#### Failed Tests (3/31056):
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9710/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest_with_topology_changes___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [gw1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9710/testReport/junit/%28root%29/%28empty%29/Tests___dtest___gw1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+gw1)
- [gw4](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9710/testReport/junit/%28root%29/%28empty%29/Tests___dtest___gw4) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+gw4)
#### Build Details:
- Duration: 3 hr 20 min
- Builder: spider4.cloudius-systems.com

piodul commented:
The test `test_hintedhandoff_switch_config_in_runtime_via_http_api` is failing too frequently. There is no use in fighting CI, we should fix it before merging the PR.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9738/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9738/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: topology_experimental_raft/test_fencing
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9738/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10133/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2110/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9738/)
#### Failed Tests (2/31456):
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9738/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
- [test_hintedhandoff_switch_config_in_runtime_via_http_api](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9738/testReport/junit/hintedhandoff_additional_test/TestHintedHandoff/Tests___dtest_with_topology_changes___test_hintedhandoff_switch_config_in_runtime_via_http_api) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_hintedhandoff_switch_config_in_runtime_via_http_api)
#### Build Details:
- Duration: 2 hr 58 min
- Builder: spider5.cloudius-systems.com

piodul commented:
> The test `test_hintedhandoff_switch_config_in_runtime_via_http_api` is failing too frequently. There is no use in fighting CI, we should fix it before merging the PR.

@dawmd It turns out that the test failure is due to the change in the name of the metric. `test_hintedhandoff_switch_config_in_runtime_via_http_api` is sensitive to the metric name and is also affected by the choice of the new names. Please adjust the test first so that it will work with the old name and the new name of the metric, then we can continue with this PR.

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10135/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10135/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: topology_experimental_raft/test_fencing
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10135/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2408/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10431/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10135/)
#### Build Details:
- Duration: 11 hr
- Builder: i-03db8a236ed15c24b (m5ad.8xlarge)

piodul commented:
@dawmd why did you choose the `backport/6.0` label specifically?

piodul commented:
Queued
